src/model/neuroevolution/individual.py

```diff
@@ -0,0 +1,220 @@
+
+        :param individual: Individual (neural network) to be modified.
+        :return: None
+        """
+        random_layer = individual.get_random_layer()
+        random_index = individual.get_random_bias_index(random_layer)
+        # TODO discuss new bias value's range, update method documentation
+        random_layer.biases[random_index] = np.random.uniform(-1, 1)
+        pass
+
+    @staticmethod
+    def change_weight_sign_mutation(individual: "ChildIndividual") -> None:
+        """
+        Changes a sign of randomly chosen weight in the neural network.
+
+        :param individual: Individual (neural network) to be modified.
+        :return: None
+        """
+        random_layer = individual.get_random_layer()
+        random_index = individual.get_random_weight_index(random_layer)
+        random_layer.weights[random_index] = -random_layer.weights[random_index]
+
+    @staticmethod
+    def multiply_neuron_weights_mutation(individual: "ChildIndividual") -> None:
+        """
+        Multiplies all weights for single randomly chosen neuron,
+        from randomly chosen layer, by random numbers from the range [0.5, 1.5].
+        There is a random multiplier chosen for every single weight.
+        :param individual: Individual to be changed.
+        :return: None
+        """
+        random_layer = individual.get_random_layer()
+        random_neuron_index = np.random.randint(0, random_layer.weights.shape[0])
+        multipliers = np.random.uniform(
+            low=0.5, high=1.5, size=(random_layer.weights.shape[1])
+        )
+        random_layer.weights[random_neuron_index] *= multipliers
+
+    @staticmethod
+    def random_neuron_weights_mutation(individual: "ChildIndividual") -> None:
+        """
+        Changes all weights for single randomly chosen neuron from randomly chosen layer.
+        :param individual: Individual to be changed.
+        :return: None
+        """
+        random_layer = individual.get_random_layer()
+        random_neuron_index = np.random.randint(0, random_layer.weights.shape[0])
+        # TODO decide the range of new random weights
+        random_layer.weights[random_neuron_index] = np.random.random(
+            (random_layer.weights.shape[1])
+        )
+
+
+ChildIndividual.available_mutations = [
+    ChildIndividual.random_weight_mutation,
+    ChildIndividual.random_bias_mutation,
+    ChildIndividual.change_weight_sign_mutation,
+    ChildIndividual.multiply_neuron_weights_mutation,
+    ChildIndividual.random_neuron_weights_mutation,
+]
+
+
+@dataclass
+class Individual:
+    neural_network: NeuralNetwork
+    adaptation: float
+
+    available_reproductions: ClassVar[
+        List[
+            Callable[
+                ["Individual", "Individual"], Tuple[ChildIndividual, ChildIndividual]
+            ]
+        ]
+    ]
+
+    @staticmethod
+    def weight_swap_reproduction(
+        father1: "Individual", father2: "Individual", weights_to_be_swapped: int = 1
```

Comment:
You both need tutorial about how reproduction works

Reply:
Everyone deserves to be happy